packages/material-ui/src/TextField/TextField.js

```diff
@@ -107,10 +107,11 @@ const TextField = React.forwardRef(function TextField(props, ref) {
       InputMore.notched = InputLabelProps.shrink;
     }
     if (label) {
+      const displayRequired = InputLabelProps?.required ?? required;
```

Comment:
This line transforms into:

```jsx
var _InputLabelProps$requ, _InputLabelProps;

var displayRequired = (_InputLabelProps$requ = (_InputLabelProps = InputLabelProps) === null ||
_InputLabelProps === void 0 ? void 0 : _InputLabelProps.required) !== null
&& _InputLabelProps$requ !== void 0 ? _InputLabelProps$requ : required;
```

What do you think of?

```suggestion
      const displayRequired = InputLabelProps && InputLabelProps.required !== null ? required : undefined;
```

Reply:
Let's stick with the correct semantics and improve transpiling with modern bundles.

Reply:
I wonder about the tradeoff:
- cost: what's the broader impact if we fully scale-out this syntax? +X% bundle size impact, will X be negligible? I wonder about why the transpiled code tests for strict equality with undefined and null vs `!= null`. Maybe Babel can/will optimize it. I hope it won't, at scale, not have the same overhead has the `class` syntax has.
- value: what cases does the new syntax allow us to better cover?

Reply:
> what's the broader impact if we fully scale-out this syntax? 

Valid point. But we need to be careful that we don't fall into worrying about things we don't need (see "yagni").

I'm looking through all our plugins and see if we can safely enable loose mode in preset-env. For only these two syntaxes https://github.com/babel/babel/issues/6978 would help.